File: app/src/main/java/com/omar/retromp3recorder/app/di/UseCaseModule.java
package com.omar.retromp3recorder.app.di;

import com.omar.retromp3recorder.app.recording.usecase.CheckPermissionsUC;
import com.omar.retromp3recorder.app.recording.usecase.CheckPermissionsUCImpl;

import dagger.Binds;
import dagger.Module;

@Module
public interface UseCaseModule {

    @Binds
    CheckPermissionsUC provideCheckPermissionsUC(CheckPermissionsUCImpl impl);

}

File: app/src/main/java/com/omar/retromp3recorder/app/di/UseCaseModule.kt
package com.omar.retromp3recorder.app.di

import com.omar.retromp3recorder.bl.CheckPermissionsUC
import com.omar.retromp3recorder.bl.CheckPermissionsUCImpl
import dagger.Binds
import dagger.Module

@Module
interface UseCaseModule {
    @Binds
    fun provideCheckPermissionsUC(impl: CheckPermissionsUCImpl): CheckPermissionsUC
}